### ISSUE TITLE
Allow enabling/disabling commit Signed-off-by on a per-project basis

### ISSUE THREAD

**Issue by sebthom** [eclipse-egit/egit#55]
### Description

EGit offers a global setting to automatically insert "Signed-off-by" messages.

![image](https://github.com/user-attachments/assets/c9ec9b94-0a73-433e-9466-4f6c67fac406)

I suggest to allow enabling/disabling this on a per project setting.

### Motivation

I have projects in my workspace from different contexts (opensource, inhouse, customers) etc. each has their own commit rules.
some require signed-off-by messages for other projects it is just noise.

Currently I have the setting disabled and manually amend commits. When context switching between projects I sometimes forget that and only after a PR and a failed CI run I see the issue. This takes time and is annoying.

**Comment by tomaswolf** [comment]
Wouldn't it make more sense to have this as a per-repository setting?

**Comment by sebthom** [comment]
@tomaswolf that's what I basically meant. :-) Thanks for your comment!

**Comment by vogella** [comment]
@sebthom is this issue still relevant to you?

**Comment by sebthom** [comment]
@vogella Yes this is still relevant. I use rather large workspaces because I like to use the cross reference / usage reference features of Eclipse and the global fulltext search. So I usually have OSS repos I contribute to and in-house/client repos that use these OSS projects in the same workspace.
